### ISSUE TITLE
Net::HTTP spy errors when the transaction_max_spans limit is exceeded

### ISSUE THREAD

**Issue by dnnyjns** [elastic/apm-agent-ruby#260]
The net_http spy produces an error when the transaction_max_spans limit is exceeded. I would expect this to fail silently instead.

```ruby
require 'elastic-apm'
require 'net/http'

ElasticAPM.start(
  transaction_max_spans: 0
)

ElasticAPM.with_transaction do
  Net::HTTP.get(URI('https://www.google.com'))
end

ElasticAPM.stop
```

Error:
```
NoMethodError: undefined method `id' for nil:NilClass
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/gems/2.3.0/gems/elastic-apm-2.0.1/lib/elastic_apm/spies/net_http.rb:28:in `block in request'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/gems/2.3.0/gems/elastic-apm-2.0.1/lib/elastic_apm.rb:234:in `with_span'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/gems/2.3.0/gems/elastic-apm-2.0.1/lib/elastic_apm/spies/net_http.rb:26:in `request'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/2.3.0/net/http.rb:1294:in `request_get'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/2.3.0/net/http.rb:481:in `block in get_response'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/2.3.0/net/http.rb:853:in `start'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/2.3.0/net/http.rb:584:in `start'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/2.3.0/net/http.rb:479:in `get_response'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/2.3.0/net/http.rb:456:in `get'
	from (irb):54:in `block in irb_binding'
	from /opt/boxen/rbenv/versions/2.3.6/lib/ruby/gems/2.3.0/gems/elastic-apm-2.0.1/lib/elastic_apm.rb:143:in `with_transaction'
	from (irb):53
	from /opt/boxen/rbenv/versions/2.3.6/bin/irb:11:in `<main>'
```

**Comment by mikker** [comment]
Yes, I will fix this right away. Thanks, @dnnyjns !

**Comment by mikker** [comment]
Should be fixed in `2.1.1`. Can you confirm?
